src/instructlab/sdg/pipeline.py

```diff
@@ -65,31 +65,45 @@ def generate(self, dataset) -> Dataset:
         Generate the dataset by running the pipeline steps.
         dataset: the input dataset
         """
-        for block_prop in self.chained_blocks:
-            block_name = block_prop["name"]
-            block_type = _lookup_block_type(block_prop["type"])
-            block_config = block_prop["config"]
-            drop_columns = block_prop.get("drop_columns", [])
-            drop_duplicates_cols = block_prop.get("drop_duplicates", False)
-            block = block_type(self.ctx, self, block_name, **block_config)
-
-            logger.info("Running block: %s", block_name)
-            logger.info(dataset)
-
-            dataset = block.generate(dataset)
-
-            # If at any point we end up with an empty data set, the pipeline has failed
-            if len(dataset) == 0:
-                raise EmptyDatasetError(
-                    f"Pipeline stopped: Empty dataset after running block: {block_name}"
-                )
-
-            drop_columns_in_ds = [e for e in drop_columns if e in dataset.column_names]
-            if drop_columns:
-                dataset = dataset.remove_columns(drop_columns_in_ds)
-
-            if drop_duplicates_cols:
-                dataset = self._drop_duplicates(dataset, cols=drop_duplicates_cols)
+        try:
+            for block_prop in self.chained_blocks:
+                block_name = block_prop["name"]
+                block_type = _lookup_block_type(block_prop["type"])
+                block_config = block_prop["config"]
+                drop_columns = block_prop.get("drop_columns", [])
+                drop_duplicates_cols = block_prop.get("drop_duplicates", False)
+                block = block_type(self.ctx, self, block_name, **block_config)
+
+                logger.info("Running block: %s", block_name)
+                logger.info(dataset)
+
+                dataset = block.generate(dataset)
```

Comment:
It looks like these lines capture the block-specific code, so you could localize your addition to this part, I think.

Reply:
Good point! Do you think it's worth containing the config parsing as well? It seems like the use of direct indexing (vs `.get`) for `"name"`, `"type"`, and `"config"` implies that these are all required fields, but I don't see anywhere that they're validated (though that might happen elsewhere that I haven't found yet?). If it's not done elsewhere, it may be worth doing that validation in the `__init__` to avoid getting part-way through the expensive pipeline operation before discovering configuration errors.

Reply:
we now have some config validation, at least for the configs we have in the tree.  See `make validate-pipelines`. It uses a jsonschema definition

Reply:
Ok, cool, makes sense!